cvmfs/catalog.h

```diff
@@ -231,6 +231,7 @@ class Catalog : public SingleCopy {
   Counters& GetCounters() { return counters_; }
 
   inline const CatalogDatabase &database() const { return *database_; }
+  inline       CatalogDatabase &database()       { return *database_; }
```

Comment:
I just find the 'value semantics' for a *protected* accessor less odd than dereferencing pointers to gain `value semantics` selectively.